utilities/tools/Update-NestedRoleAssignmentList.ps1

```diff
@@ -0,0 +1,146 @@
+        #####################
+        $newContent = ($nestedRoles | Out-String).TrimEnd()
+        $content = ($content -replace '(?ms)^\s+var builtInRoleNames = {.*?}', $newContent).TrimEnd()
+        Set-Content -Path $pathToFile -Value $content -Force -Encoding 'utf8'
+
+        # Return arrays
+        return $roles
+    }
+
+    end {
+        Write-Debug ('{0} exited' -f $MyInvocation.MyCommand)
+    }
+}
+
+<#
+.SYNOPSIS
+Update latest available Role Definitions in nested_roleassignments.bicep template
+
+.DESCRIPTION
+Update latest available Role Definitions in nested_roleassignments.bicep template for the given ProviderNamespace and ResourceType if specified,
+otherwise perform the update to the entire library
+
+.PARAMETER ProviderNamespace
+Optional. The Provider Namespace to fetch the role definitions for
+
+.PARAMETER ResourceType
+Optional. The ResourceType to fetch the role definitions for
+
+.EXAMPLE
+Update-NestedRoleAssignmentList
+
+Update all nested_roleassignments.bicep found in the library with latest available Role Definitions
+
+.EXAMPLE
+Update-NestedRoleAssignmentList -ProviderNamespace 'Microsoft.KeyVault' -ResourceType 'vaults'
+
+Update nested_roleassignments.bicep template for [Microsoft.KeyVault/vaults] module with latest available Role Definitions
+#>
+function Update-NestedRoleAssignmentList {
+
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    param(
+        [Parameter(Mandatory = $false)]
+        [string] $ProviderNamespace,
+
+        [Parameter(Mandatory = $false)]
+        [string] $ResourceType
+    )
+
+    begin {
+        Write-Debug ('{0} entered' -f $MyInvocation.MyCommand)
+        $repoRootPath = (Get-Item $PSScriptRoot).Parent.Parent
+        $modulesPath = Join-Path $repoRootPath 'modules'
+        $FileToSearch = 'nested_roleAssignments.bicep'
+    }
+
+    process {
+        if ($ProviderNamespace -and $ResourceType) {
+            ########################################
+            ## Update RBAC roles for single module #
+            ########################################
+            if ($PSCmdlet.ShouldProcess("Role Assignments for module [$ProviderNamespace/$ResourceType]", 'Update')) {
+                $null = Update-NestedRoleAssignmentListInner -ProviderNamespace $ProviderNamespace -ResourceType $ResourceType -Verbose
+            }
+        } else {
+            ############################################
+            ## Update RBAC roles for the whole library #
+            ############################################
+            Set-Location $modulesPath
+            $searchFile = Join-Path $modulesPath '**' $FileToSearch
+            $rbacPathList = Get-ChildItem -Path $searchFile -Recurse
+            foreach ($item in $rbacPathList) {
+                $FullFilePath = $item.FullName
+                $relativeFilePath = Get-Item $FullFilePath | Resolve-Path -Relative
+                $relativeFilePath = $relativeFilePath -replace '\.\\', ''
```

Comment:
I wonder - all this replaces require the the slashes to be Windows-based. If think to be on the save side, it may make sense to change the implementation to always change `'\'` to `'/'` - and then process that.

For example:
```bicep
($relativeFilePath -replace '\\', '/') -replace '\.\/', ''
```

Reply:
We ust this a lot in other utilities too

Reply:
updated and retested